src/main/java/frc/robot/subsystems/snapback/hood/HoodIOSim.java

```diff
@@ -1,3 +1,64 @@
 package frc.robot.subsystems.snapback.hood;
 
-public class HoodIOSim {}
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.controller.SimpleMotorFeedforward;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.system.plant.LinearSystemId;
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.wpilibj.simulation.DCMotorSim;
+import frc.robot.constants.Constants;
+
+public class HoodIOSim implements HoodIO {
+  private final DCMotorSim motorSim;
+  private double appliedVolts = 0.0;
+
+  private ProfiledPIDController controller;
+  private SimpleMotorFeedforward feedforward;
+
+  public HoodIOSim() {
+    motorSim =
+        new DCMotorSim(
+            LinearSystemId.createDCMotorSystem(
+                HoodConstants.HOOD_GEARBOX, 0.004, HoodConstants.GEAR_REDUCTION),
+            HoodConstants.HOOD_GEARBOX,
+            0.004);
+    controller =
+        new ProfiledPIDController(
+            HoodConstants.GAINS.kp(),
+            HoodConstants.GAINS.ki(),
+            HoodConstants.GAINS.kd(),
+            new TrapezoidProfile.Constraints(HoodConstants.GAINS.kv(), HoodConstants.GAINS.ka()));
```

Comment:
kv and ka are the gains for the feedforward of the controller, not the constraints of the motion profile. We need two more constants for the max velocity and max acceleration of the hood as it moves.